### ISSUE TITLE
Comments from includes are not removed

### ISSUE THREAD

**Issue by penibelst** [penibelst/jekyll-compress-html#34]
A user asks via email:

> I installed "jekyll-compress-html 1.1.1" a few days ago.

> I am currently working on a project involving landing pages for mobile.

> This is why I choose to use Jekyll and all possible minifying procedures to get pages as lightweight as possible.

> So, here's what I just discovered a few hours ago using compress-html:

>  My _config.yml has the "Full-blown sample" you suggested on your site

```yaml
compress_html:
  clippings: all
  comments: ["<!-- ", " -->"]
  endings: all
  ignore:
    envs: [local]
```

> It removes perfectly the comments [""] if they are initially contained in the source layout file saved in _layouts.

> But if a layout contains includes {% include snippet-to-include.html %}, the comments [""] inside these includes are not removed in the final output.

> I just discovered that, as I have a simple landing page layout where I include a "survey" snippet: I had initially this survey in English, and then I got it translated in German, and decided to keep both versions in the same include file. (But commenting out the English version, as I only need the German one now).

> I use the latest jekyll (v.2).
> My guess is it's a question of order in the minifying process.
> But my knowledge stops here on this subject.

**Comment by penibelst** [comment]
Please provide the full content of the include.

**Comment by eric-zh** [comment]
I have a layout (saved in _layouts) where I include "snippet1.html" (saved in _includes)

Inside this "snippet1.html", I have another include "snippet2.html" (saved in _includes too) (a bit of a Russian Doll).

Here the is content of the "snippet1.html"
____________________________________
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse interdum, erat in ullamcorper fringilla, dolor nisl facilisis nibh, in semper eros magna ut dolor.
 {% include snippet2.html %}


____________________________________


Technically, the output of the page works fine as expected.

Looking at the source file, the comment are is compressed, but it is NOT removed.

If I delete the  {% include snippet2.html %} in the commented area, then all the comment area is removed (which is expected)

**Comment by penibelst** [comment]
@eric-zh To format your example you must enclose it with three backticks "```".

**Comment by penibelst** [comment]
In #35 I try to reproduce your issue without success. Everything works as expected on my local machine and on Travis. Can you show me your full source? What is in the snippet 2 at least?

**Comment by penibelst** [comment]
@eric-zh Any progress?

**Comment by eric-zh** [comment]
Anatol, 
Here's (finally!) an example set that I stripped a bit of useless elements. So the end result may not have the usual functionality (put the URL of a youtube video in YAML part of the post, and it will automatically insert the embed code in the index.html page).

in the directory _includes, I have the file "include-video.html"

```
<!-- 
vid_poster: /assets/videos/mobile180p.jpg
vid_id: ID72186958
vid_youtube: https://www.youtube.com/watch?v=fKR7_ZxdE_o
-->

<!-- START OF THE PLAYER EMBEDDING TO COPY-PASTE -->
<script src="{{ BASE_PATH }}/player/youtube.min.js"></script>

<style media="all" type="text/css">
    .video-js {/*padding-top: 56.25%; */padding-top: 42.6%;}/*16:9 or 16:7*/
	.vjs-fullscreen {padding-top: 0px}
</style>

<video x-webkit-airplay="allow" id="{{ page.vid_id }}" class="video-js vjs-default-skin vjs-big-play-centered"
	  	  data-setup='{"ytcontrols": false, "muted":true, "loop": false, "autoplay": false, "techOrder": ["youtube"], "quality": "240p", "src": "{{ page.vid_youtube }}"}'>

	 <p class="vjs-no-js">Your browser does not support the video tag.</p>
</video>

<!-- back to 0 with poster image -->
<script type="text/javascript">
	var video = videojs('{{ page.vid_id }}');video.on('ended', function() {video.trigger('loadstart');video.posterImage.show();});
</script>

<!-- END OF THE PLAYER EMBEDDING -->
```

in the directory _layouts, I have the file "Layout-example.html"
```
---
layout: compress
---

<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">

        <title>Video {{ page.title }}</title>

      {% include meta.html %}
      
      {% include css_online.html %}
      <link href="{{ BASE_PATH }}/assets/themes/style.css" rel="stylesheet">

      <!-- for video player -->
   
      <!--<style media="all" type="text/css">@font-face{font-family:VideoJS;src:url(//vjs.zencdn.net/f/3/vjs.eot);src:url(//vjs.zencdn.net/f/3/vjs.eot?#iefix) format('embedded-opentype'),url(//vjs.zencdn.net/f/3/vjs.woff) format('woff'),url(//vjs.zencdn.net/f/3/vjs.ttf) format('truetype'),url(//vjs.zencdn.net/f/3/vjs.svg#icomoon) format('svg');font-weight:400;font-style:normal}</style>-->

      <style media="all" type="text/css">@font-face{font-family:VideoJS;src:url(//vjs.zencdn.net/f/3/vjs.woff) format('woff'),url(//vjs.zencdn.net/f/3/vjs.ttf) format('truetype');}</style>
      
      <link href="//vjs.zencdn.net/4.12/video-js.css" rel="stylesheet">
      <script src="//vjs.zencdn.net/4.12/video.js"></script>
      
      <!-- for video player -->

    </head>

    <body style="">

        <div class="static-mobile-image-container">
            {% include include-video.html %}
            <div class="container">
                <h2 style="">{{ page.title }}</h2>
            </div>
        </div>

        <!-- {% include optin-form1.html %} -->

        <!-- {% include JB/analytics %} -->
        {% include js_online.html %}

    </body>
</html>
```

in the directory _posts, I have the file "Layout-example.html"
```
---
layout : Layout-example
category : lessons
tags : [intro, beginner, jekyll, tutorial]
title: debugging
creation: 24.01.2015 - 09.58 GMT

vid_poster: /assets/videos/mobile180p.jpg
vid_id: ID72186958
vid_youtube: https://www.youtube.com/watch?v=fKR7_ZxdE_o
---
In in est ac nisi feugiat malesuada. Nullam lacinia volutpat nunc, sit amet rhoncus ex sollicitudin sit amet. Mauris ut lorem est. Duis tempus aliquam nunc, eu imperdiet tellus efficitur in.
```

The output file "index.html" still contains:
- some commented parts from the <head> area of the "Layout-example.html"
- some commented parts of the "Layout-example.html" (+comments from the overriding CSS I inserted)

I understand the code is heavily commented and this creates an unusual case!
Thank you for your help.

**Comment by penibelst** [comment]
This is a huge example. I will need time to go through everything.

I can say immediately, that my layout follows the comment option exactly. The whitespace

* after the open comment tag 
* and before the close tag

means only such comments are stripped.

```yaml
compress_html:
  comments: ["<!-- ", " -->"]
#                 ^    ^
```

Your source contains mixed comment style. So this could be the cause for the issue.

**Comment by penibelst** [comment]
Okay, I see the issue. One part doesn’t match your comment style, so it’s not removed. This is the expected behavior.

```html
<!--<style media="all" type="text/css">@font-face{font-family:VideoJS;src:url(//vjs.zencdn.net/f/3/vjs.eot);src:url(//vjs.zencdn.net/f/3/vjs.eot?#iefix) format('embedded-opentype'),url(//vjs.zencdn.net/f/3/vjs.woff) format('woff'),url(//vjs.zencdn.net/f/3/vjs.ttf) format('truetype'),url(//vjs.zencdn.net/f/3/vjs.svg#icomoon) format('svg');font-weight:400;font-style:normal}</style>-->
```

If you insert whitespaces around comment tags, the comment will be stripped.

The other part must be removed, but it’s not.

```html
<!-- 
vid_poster: /assets/videos/mobile180p.jpg
vid_id: ID72186958
vid_youtube: https://www.youtube.com/watch?v=fKR7_ZxdE_o
-->
```

I will investigate here.

**Comment by eric-zh** [comment]
Thank you Anatol. 

There are also these comments in the overriding CSS "/* */"
Wouldn't it possible to define the different types of comments in the _config.yml? 
Something like:
comments: [""], ["/*", "*/"]

**Comment by kleinfreund** [comment]
Compressing CSS is a completely different task. HTML and CSS are _really_ different. If anything, this would make up for a new project. Technically, that could work with a layout as well as long as the CSS files have front matter.

**Comment by penibelst** [comment]
@eric-zh As @kleinfreund said the compress layout is not for CSS. Jekyll supports [Sass](http://sass-lang.com/) out of the box, which includes a compression algorithm.

I don’t want to support different comment types. This would be an overkill.

**Comment by penibelst** [comment]
The bug is fixed in 1.1.2.
